### PR TITLE
Zarina Sept post on search

### PR DESCRIPTION
**Pull request by zmustapha**

### CODE REVIEW

File: content/articles/searching-1.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
+---
+title: "Considerations for Search"
+date: "2018-09-19"
+type: "post"
+author_name:  "Zarina Mustapha"
+author_url: "https://ctl.columbia.edu/about/team/mustapha/"
+lede: "Search functionality requires a thoughtful deliberation so that the
+model, UI, and UX can best serve the users’s needs. This post lists some useful
```
Comment (sdreher) [inline content/articles/searching-1.md]: extra apostrophe s here.

File: content/articles/searching-1.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+### What is the user searching for? ###
+To put simply, the user is searching for an object. But what attributes define
+the object? Are there relationships with other objects? Are these relationships
+relevant to the search? In the “Green Notebook Quest” story, the object that I
+was looking for can initially be broken down as follows:  
+`type: notebook`  
+`color: green`
+
+And as it turned out, there were more than one green notebook.
```
Comment (sdreher) [inline content/articles/searching-1.md]: there `was`

File: content/articles/searching-1.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+for, or title, (“Math”), and tags (“cat sticker”): 
+
+<div class="row">
+<div class="col-sm-4 text-center">
+<img src="/img/assets/notebook-lg.png">
+<p><i class="fa fa-times-circle fa-3x text-muted" aria-hidden="false"></i></p>
+</div>
+<div class="col-sm-8">
+<code>
+type: notebook<br />
```
Comment (sdreher) [inline content/articles/searching-1.md]: Adding commas between the search criteria might help to distinguish them visually for the reader

File: content/articles/searching-1.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+tag: cat<br />
+size: "small"<br />
+width: 7in<br />
+height: 11in
+</code>
+</div>
+</div>
+
+From this example, we see that it is helpful to understand the definition of
+the object the is searching for, in any given application, so that we can model
```
Comment (sdreher) [inline content/articles/searching-1.md]: we see that it is helpful to understand the definition of the object `we are searching for` or `the user is searching for`.

File: content/articles/searching-1.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+overall experience of the site? How does it add to, or impede the user’s
+learning experience? Is the search functionality a solution to poor information
+architecture, or a way to address a complex site model? Do we need to track the
+search patterns? This set of questions can provide insights on what the search
+should do, anticipate user search patters, and shape the expected UX around it.
+
+### How is search being done? ###
+When we have user stories and requirements written on the search feature, we
+can now think about how to implement them. What search engines will we use?
+ComplilED, for example, uses [lunr.js](https://lunrjs.com) because this blog is
```
Comment (sdreher) [inline content/articles/searching-1.md]: `Compiled`

File: content/articles/searching-1.md [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+a Hugo-generated static site with small dataset, and we decided that full-text
```
Comment (sdreher) [inline content/articles/searching-1.md]: with `a` small dataset

File: content/articles/searching-1.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+search is good enough. For the mapping project
+_[Writ Large](https://writlarge.ctl.columbia.edu/map/)_,
+a Django project with a manageable size of data set, we use Django PostgreSQL
+search tools. We decided to go with
+[Solr](http://lucene.apache.org/solr/) for
+[Footprints](https://footprints.ccnmtl.columbia.edu/search/) because there are
+thousands of records to go through, using much more robust faceted search.
+
+However, considerations for search doesn’t simply stop with the technology.
```
Comment (sdreher) [inline content/articles/searching-1.md]: However, considerations for search `don't` simply

File: content/articles/searching-1.md [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+
+So now, we have to decide how to design the UI and UX around this simple search
+pattern. Can we assume that the user has a certain knowledge of superuser
+searches so we can condense everything into one search field, i.e.
+`type:notebook AND color:green AND size:"small"` etc? Or should we design an
+elegant search interface that appeals more to the user’s intuition? We did the
+latter for [Footprints](https://footprints.ccnmtl.columbia.edu/search/) search.
+
+### Where is the user searching? ###
+The may be instances where we set limits to user searches (search only
```
Comment (sdreher) [inline content/articles/searching-1.md]: `There` may be

File: content/articles/searching-1.md [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+published or public records if user is not an admin, for example). What fields
+or attributes are being exposed to the user search? The users may want to set
+their own limits to the search as well. The latter case can also be thought as
+a facet of the search. The areas of search in “Green Notebook Quest,” for
+example, were “shelves by the cabinet in my room”
+
+Why is this important? As always, for UI, and UX.
+
+### Who can see what the user is looking for? ###
+Sometimes, we show to hide certain search results based on user permissions.
```
Comment (sdreher) [inline content/articles/searching-1.md]: Sometimes, `we hide certain`

File: content/articles/searching-1.md [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+Only editors will see unpublished articles, for example. This type of decisions
```
Comment (sdreher) [inline content/articles/searching-1.md]: This type of `decision`

File: content/articles/searching-1.md [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+needs to be included in testing so we can be certain that a zero match is
+indeed and intended match.
```
Comment (sdreher) [inline content/articles/searching-1.md]: a zero match is indeed `an` intended match

File: content/articles/searching-1.md [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+
+### How can the user see what the user looking for? ###
+We ask this question so we can design the search result page. Should the
+results page be separate? For
+[Footprints](https://footprints.ccnmtl.columbia.edu/search/), we decided to
+show all records at first, and then show only records that match the search
+queries. This blog’s simple search displays mathing posts in a pop-up panel. In
```
Comment (sdreher) [inline content/articles/searching-1.md]: simple search displays `matching` posts

File: content/articles/searching-1.md [hunk 13]
```diff
@@ -0,0 +1,257 @@
...
+_[Writ Large](https://writlarge.ctl.columbia.edu/map/)_, all pins that don’t
+match the search criteria are dimmed instead of hidden.
+
+### The result from the user’s search, does it match what the user is looking
```
Comment (sdreher) [inline content/articles/searching-1.md]: This line is not displaying properly in the preview. Possibly a CRLF after `looking`?

File: content/articles/searching-1.md (hunk 1 above)
Comment (zmustapha) [reply content/articles/searching-1.md]: Fixed

File: content/articles/searching-1.md [hunk 14]
```diff
@@ -200,27 +200,33 @@ example, were “shelves by the cabinet in my room”
...
 The search implementation is incomplete without this verification, and we do
 this through testing. Write a test. We set up a search against a known set of
 data. We test using different search parameters. We test it against different
 user permission levels. Test, verify, and so on.
 
+### To summarize... ###
+The search feature is not just about the engines looking for things in the
+database. Search begins with the user’s intention, and it revolves around the
+user’s expectations in searching. The search experience should be intuitive,
+as well as informative. I hope he questions here can serve as a starting guide to a
```
Comment (sdreher) [inline content/articles/searching-1.md]: I hope `the` questions here.

Good summary. Like it.
